modules/rostests/apitests/kernel32/Pipes.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         GNU GPLv2 only as published by the Free Software Foundation
+ * PURPOSE:         Test for pipe (CORE-17376)
+ * PROGRAMMER:      Simone Mario Lombardo
+ */
+
+#include "precomp.h"
+
+static const char* g_PipeName = "\\\\.\\pipe\\rostest_pipe";
+
+#define MINBUFFERSIZE 1
+#define MAXBUFFERSIZE 255
+#define TEST_MESSAGE "Test"
+#define TEST_MESSAGE_SIZE 4
+
+static DWORD g_dwReadBufferSize;
+
+void CheckError(BOOL Success, DWORD dwLastError, const char* errorMsg)
+{
+    ok(Success, errorMsg);
+    if (!Success)
+    {
+        trace("Error: %s. LastError: 0x%lX\n", errorMsg, dwLastError);
+    }
+}
+
+DWORD WINAPI PipeWriter(_In_ PVOID Param)
+{
+    HANDLE hPipe = (HANDLE)Param;
+    DWORD cbWritten = 0;
+    BOOL Success = WriteFile(hPipe, TEST_MESSAGE, TEST_MESSAGE_SIZE, &cbWritten, NULL);
+    DWORD dwLastError = GetLastError();
+
+    CheckError(Success, dwLastError, "Pipe's WriteFile failed");
+    ok(cbWritten == TEST_MESSAGE_SIZE, "Invalid cbWritten: 0x%lx\n", cbWritten);
+
+    return 0;
+}
+
+DWORD WINAPI PipeReader(_In_ PVOID Param)
+{
+    HANDLE hPipe = (HANDLE)Param;
+    CHAR* szOutMsg = (CHAR*)malloc(g_dwReadBufferSize);
```

Comment:
`CHAR szOutMsg[MAXBUFFERSIZE]` seemed better to me.
Also, this is no "sz", just a buffer.